spec/ruby/core/process/egid_spec.rb

```diff
@@ -15,5 +15,39 @@
 end
 
 describe "Process.egid=" do
-  it "needs to be reviewed for spec completeness"
+
+  platform_is_not :windows do
+    it "raises TypeError if not passed an Integer or String" do
+      -> { Process.egid = Object.new }.should raise_error(TypeError)
+    end
+
+    it "sets the effective group id to its own gid if given the username corresponding to its own gid" do
+      raise unless Process.gid == Process.egid
+
+      require "etc"
+      group = Etc.getgrgid(Process.gid).name
+
+      Process.egid = group
+      Process.egid.should == Process.gid
+    end
+
+    as_user do
+      it "raises Errno::ERPERM if run by a non superuser trying to set the root group id" do
+        -> { Process.egid = 0 }.should raise_error(Errno::EPERM)
+      end
+
+      it "raises Errno::ERPERM if run by a non superuser trying to set the group id from group name" do
+        -> { Process.egid = "root" }.should raise_error(Errno::EPERM)
+      end
+    end
+
+    as_superuser do
+      context "when ran by a superuser" do
+        it "sets the effective group id for the current process if run by a superuser" do
+          Process.egid = 1
```

Comment:
This seems way too dangerous as a side-effect, could you restore the original `egid` in `ensure`?

Reply:
I'll do it, to avoid another feedback cycle.